src/Application/Actions/Donations/Update.php

```diff
@@ -113,6 +113,20 @@ protected function action(Request $request, Response $response, array $args): Re
                     );
                 }
 
+                if ($donationData->autoConfirmFromCashBalance && $donation->getPaymentMethodType() !== 'customer_balance') {
+                    $this->entityManager->rollback();
+
+                    // Log a warning to more easily spot occurrences in dashboards.
+                    $this->logger->warning(
+                        "Donation ID {$args['donationId']} auto-confirm attempted with '{$donation->getPaymentMethodType()}' payment method",
+                    );
+
+                    return $this->validationError($response,
+                        "Donation ID {$args['donationId']} could not be auto-confirmed",
+                        'Processing incomplete. Please refresh and check your donation funds balance'
```

Comment:
If this indicates a bug in the client maybe the message needs to be different, e.g. 
```suggestion
                        "Processing incomplete. Auto-confirm is not possible for a '{$donation->getPaymentMethodType()}' donation"
```